src/coreclr/inc/corprof.idl

```diff
@@ -4308,6 +4312,33 @@ interface ICorProfilerInfo14 : ICorProfilerInfo13
                 [out] EVENTPIPE_PROVIDER        *pProvider);
 }
 
+[
+    object,
+    uuid(B446462D-BD22-41DD-872D-DC714C49EB56),
+    pointer_default(unique),
+    local
+]
+interface ICorProfilerInfo15 : ICorProfilerInfo14
+{
+    /*
+     * EnumerateGCHeapObjects is a method that iterates over each object in the GC heap.
+     * For each object, it invokes the provided callback function which should return a bool
+     * indicating whether or not enumeration should continue.
+     *
+     * Parameters:
+     * - callback: A function pointer to the callback function that will be invoked for each object in the GC heap.
+     *            The callback function should accept an ObjectID and a void pointer as parameters and return a BOOL.
+     * - callbackState: A void pointer that can be used to pass state information to the callback function.
+     *
+     * Returns:
+     * - HRESULT: A code indicating the result of the operation. If the method succeeds,
+     *           it returns S_OK. If it fails, it returns an error code.
+     */
+    HRESULT EnumerateGCHeapObjects(
+        [in] object_callback callback,
```

Comment:
`object_callback` isn't defined in corprof.idl, I think the build happens to work because it's defined in some other runtime header. We should keep corprof.h self contained as much as possible and provide the definition here. It should be the same definition as `object_callback` but with a new name so it doesn't clash

Reply:
I added it in the typedef section at L865, so it is pretty far from where its used. Would it be preferred to move it to right above the interface scope just to keep it close?

I think from a `git grep -w object_callback` in the branch, there doesn't seem to be any other instances besides `corprof.idl`, `corprof.h`, and using it in `proftoeeinterfaceimpl.cpp/.h`

Reply:
Oh, I didn't see the definition for some reason. It's fine to leave it where it is but can you change the name to match the style of other callbacks? Something like `ObjectCallback`.

I got it confused with `object_callback_func` in `gcpriv.h

Reply:
Changed to ObjectCallback, to be similar casing to the other callback ObjectReferenceCallback